### PR TITLE
#17: Omit Shopware base packages in plugin .zip

### PR DESCRIPTION
**Pull request by moehrenzahn**
There already is some logic to strip the shopware packages from the composer.json in FroshPluginUploader\Components\PluginZip::needComposerToRun, it would be easy to update the temporary composer.json so the Shopware sources are not added to the package during composer install.

See #17.

### CODE REVIEW

**Comment by shyim** [comment]
Did you checked that the shopware packages are again in require in the zip content?

**Comment by moehrenzahn** [comment]
No, they are not anymore. Is that a requirement?

From my perspective it makes sense that in the `.zip` the `composer.json` matches the `vendor` dir, so if there are no Shopware base packages in `vendor`, they should not be in `composer.json`

**Comment by shyim** [comment]
The require is also there to check are the components there, which this plugin requires. When you remove all shopware/* also the store upload does not work

**Comment by moehrenzahn** [comment]
Good to know! I'll take a look at it.

**Comment by moehrenzahn** [comment]
Thank you for your feedback! I modified my PR to preserve the original composer.json in the .zip package

File: src/Components/PluginZip.php [hunk 1]
```diff
@@ -31,9 +31,17 @@ public function zip(string $directory, ?string $branch = null): string
...
         $this->exec(sprintf('git archive %s | tar -x -C %s', escapeshellarg($branch), escapeshellarg($pluginName)));
 
-        // Install composer dependencies
-        if ($this->needComposerToRun($directory)) {
-            $this->exec('composer install --no-dev -n -o -d ' . escapeshellarg($pluginName));
+        $composerJson = $directory . '/' . $pluginName . '/composer.json';
+        $composerJsonBackup = $directory . '/' . $pluginName . '/composer.json';
+
+        if (file_exists($composerJson)) {
+            copy($composerJson, $composerJsonBackup);
```
Comment (shyim) [inline src/Components/PluginZip.php:38]: Are the both files not the same?

Comment (moehrenzahn) [reply src/Components/PluginZip.php:38]: 😱 You are correct, I forgot the most important thing! Should be fixed now.

File: src/Components/PluginZip.php [hunk 2]
```diff
@@ -92,14 +101,26 @@ private function exec(string $command): void
...
+        return true;
+    }
+
+    /**
+     * Remove Shopware base packages from composer.json
+     * so they aren't bundled with the plugin.
+     *
+     * @param string $composerJsonPath
+     */
+    private function filterShowpareDependencies(string $composerJsonPath): void
```
Comment (shyim) [inline src/Components/PluginZip.php:122]: Typo in function name 😂

File: src/Components/PluginZip.php [hunk 3]
```diff
@@ -31,9 +31,18 @@ public function zip(string $directory, ?string $branch = null): string
...
+
+        if (file_exists($composerJson)) {
+            copy($composerJson, $composerJsonBackup);
+            $this->filterShowpareDependencies($composerJson);
+            // Install composer dependencies
+            if ($this->needComposerToRun($composerJson)) {
+                $this->exec('composer install --no-dev -n -o -d ' . escapeshellarg($pluginName));
+            }
+
+            rename($composerJsonBackup, $composerJson);
```
Comment (shyim) [inline src/Components/PluginZip.php:45]: Does this rename work? Cause the $composerJson is not deleted? 🤔

Comment (moehrenzahn) [reply src/Components/PluginZip.php:45]: I just added the "rename" literally a minute ago. You probably still have "copy" checked out.

File: src/Components/PluginZip.php (hunk 2 above)
Comment (moehrenzahn) [reply src/Components/PluginZip.php]: 😶 now fixed

**Pull request merged**